discovery_engine_core/web-api/src/handlers.rs

```diff
@@ -302,3 +305,67 @@ pub(crate) async fn handle_delete_document_properties(
         }
     }
 }
+
+#[instrument(skip(state))]
+pub(crate) async fn handle_get_document_property(
+    doc_id: DocumentId,
+    prop_id: DocumentPropertyId,
+    state: Arc<AppState>,
+) -> Result<Box<dyn Reply>, Infallible> {
+    match state.elastic.get_document_property(&doc_id, &prop_id).await {
+        Ok(Some(property)) => Ok(Box::new(DocumentPropertyResponse::new(property).to_reply()) as _),
+        Ok(None) => Ok(Box::new(StatusCode::NOT_FOUND) as _),
+        Err(Error::Elastic(error)) if matches!(error.status(), Some(StatusCode::NOT_FOUND)) => {
```

Comment:
:art: ? I think it might be a bit nicer if we would handle elastic not found errors as part of `get_document_property`.